msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AbstractApplicationBase.java

```diff
@@ -0,0 +1,476 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+import org.slf4j.Logger;
+
+import javax.net.ssl.SSLSocketFactory;
+import java.net.MalformedURLException;
+import java.net.Proxy;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+
+import static com.microsoft.aad.msal4j.ParameterValidationUtils.validateNotBlank;
+import static com.microsoft.aad.msal4j.ParameterValidationUtils.validateNotNull;
+
+/**
+ * Abstract class containing common methods and properties for {@link PublicClientApplication},
+ * {@link ConfidentialClientApplication}, and {@link ManagedIdentityApplication}
+ */
+public abstract class AbstractApplicationBase implements IApplicationBase {
+
+    protected Logger log;
+    protected Authority authenticationAuthority;
+
+    @Accessors(fluent = true)
+    @Getter(AccessLevel.PACKAGE)
+    protected TokenCache tokenCache;
+
+    @Accessors(fluent = true)
+    @Getter
+    private String correlationId;
+
+    @Accessors(fluent = true)
+    @Getter
+    private boolean logPii;
+
+    @Accessors(fluent = true)
+    @Getter
+    private Proxy proxy;
+
+    @Accessors(fluent = true)
+    @Getter
+    private SSLSocketFactory sslSocketFactory;
+
+    @Accessors(fluent = true)
+    @Getter(AccessLevel.PACKAGE)
+    private ServiceBundle serviceBundle;
+
+    @Accessors(fluent = true)
+    @Getter(AccessLevel.PACKAGE)
+    private Consumer<List<HashMap<String, String>>> telemetryConsumer;
+
+    @Accessors(fluent = true)
+    @Getter
+    private IHttpClient httpClient;
+
+    @Accessors(fluent = true)
+    @Getter
+    private Integer connectTimeoutForDefaultHttpClient;
+
+    @Accessors(fluent = true)
+    @Getter
+    private Integer readTimeoutForDefaultHttpClient;
+
+    @Accessors(fluent = true)
+    @Getter
+    private String applicationName;
```

Comment:
Please leave `appname` and `appver` out, as we're not hitting ESTS. 

PS: they are not actually used :)

Reply:
Moved it back to AbstractClientApplicationBase in the latest commit (they were referenced in the RequestContext class which now uses AbstractApplicationBase, so I figured they could've been relevant for managed identity).

However, since they aren't actually used should we start marking them as deprecated? (in a different PR)